src/marks/index.js

```diff
@@ -74,10 +77,69 @@ class Strike extends TipTapStrike {
 
 }
 
+class Link extends TipTapLink {
+
+	get schema() {
+		const inherited = super.schema
+		return {
+			...inherited,
+			parseDOM: [
+				{
+					tag: 'a[href]',
+					getAttrs: dom => ({
+						href: parseHref(dom),
+					}),
+				},
+			],
+			toDOM: node => ['a', {
+				href: domHref(node),
+				rel: 'noopener noreferrer nofollow',
+			}, 0],
+		}
+	}
+
+	get plugins() {
+		if (!this.options.openOnClick) {
+			return []
+		}
+
+		return [
+			new Plugin({
+				props: {
+					handleClick: (view, pos, event) => {
+						const { schema } = view.state
+						const attrs = getMarkAttrs(view.state, schema.marks.link)
+
+						if (attrs.href && event.target instanceof HTMLAnchorElement) {
+							event.stopPropagation()
+							const htmlHref = event.target.href
+							if (event.button === 0 && !event.ctrlKey && htmlHref.startsWith(window.location.origin)) {
+								const query = OC.parseQueryString(htmlHref)
+								const fragment = OC.parseQueryString(htmlHref.split('#').pop())
+								if (query.dir && fragment.relPath) {
+									const filename = fragment.relPath.split('/').pop()
+									const path = `${query.dir}/${filename}`
+									OCA.Viewer.open({ path })
```

Comment:
This does not have the `fileId` as a fallback.
Sharing may lead to different directory layouts for different users.
If the calculated path is wrong for the current user... there is no way for the Viewer to recover from that.

We could find out the right path by looking up the file properties by id.
But that adds complexity and another roundtrip.
Since this is an optimization to open links more smoothly then with window.location it would be nice to avoid the roundtrip.

Not sure what the best approach here is.

Reply:
Hm, I'm also not sure what to do in that regards.